scripts/install-helm.sh

```diff
@@ -0,0 +1,107 @@
+#!/bin/bash
+#
+#  Copyright (c) 2024 Red Hat, Inc.
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  You may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+# Script to handle CLI helm installation for k8s and OCP
+#
+# Requires: oc or kubectl
+
+usage() {
+echo "
+This script simplifies and automates the installation process of Helm charts on Kubernetes (K8s) and OpenShift Container Platform (OCP) clusters.
+It detects whether 'oc' or 'kubectl' is installed and ensures that the user is logged into a cluster.
+The script also attempts to detect the cluster router base and updates the Helm chart configuration accordingly.
+
+Usage:
+  $0 [OPTIONS]
+
+Options:
+  --router-base <router-base>  : Manually provide the cluster router base if auto-detection fails.
+  --help                       : Show this help message and exit.
+
+Examples:
+  $0                               # Auto-detects router base and installs the Helm chart
+  $0 --router-base example.com     # Manually specifies the router base and installs the Helm chart
+"
+}
+
+# Function to check if a command exists
+command_exists() {
+    command -v "$1" >/dev/null 2>&1
+}
+
+# Check if either oc or kubectl is installed
+if command_exists oc; then
+    CLI="oc"
+elif command_exists kubectl; then
+    CLI="kubectl"
+else
+    echo "Error: Neither 'oc' nor 'kubectl' is installed. Please install one of them to proceed."
+    exit 1
+fi
```

Comment:
We should provide a flag that can be used to override the automatic detection here in case someone wants to use kubectl instead of oc even if they are both installed.

Reply:
why?

Reply:
Great Idea -  For this I have added an option cli that takes either oc or kubectl as a value. The script uses the specified cli tool if provided; otherwise, it will fall back to the automatic detection.

Reply:
yeah the "why?" was at Corey because I can't grok why if you have both tools installed you'd care which one was used to perform the cluster introspection.